bindings/javascript/packages/common/promise.ts

```diff
@@ -238,7 +273,7 @@ class Statement {
    * @param raw Enable or disable raw mode. If you don't pass the parameter, raw mode is enabled.
    */
   raw(raw) {
-    this.stmt.raw(raw);
+    this.stmt.apply(s => s.raw(raw));
```

Comment:
are any of these `.apply()` cases ever called on a stmt that is `MaybePromise` and if so, doesn't that mean that e.g. this function completes before `.raw()` is actually called on the `Statement` (plus my earlier worries about promise rejection of `lazy` crashing Node.js

Reply:
Yes. Drizzle folks asked us to make prepare and DB instance creation sync if possible motivating this by developers who can't use top level await and also I guess this is more nicely fit into their "execution flow".

So, this apply can be called for `MaybePromise` if statement was created for not-yet-connected database instance. In this case we will chain all operations and defer their execution until first real execution method.

This is not nice - but I don't have any good ideas how else to resolve that demand from Drizzle. All users who uses turso client directly must use `connect` wrapper or they can explicitly call connect and avoid any defer logic.

Reply:
Could asynchronous apply store the functions (e.g. inline, inside the closure in `maybePromise` function:

```js
function maybePromise(lazy) {
  const deferredFuncs = [];
  return {
    apply(fn) { deferredFuncs.push(fn); }
    resolve() {
      return lazy.then(val => {
        for (func of deferredFuncs) {
          func(val);
        }
        return val;
      });
    }
  }
}
```